cpp/src/Slice/Parser.cpp

```diff
@@ -3196,7 +3196,7 @@ Slice::Operation::hasMarshaledResult() const
 }
 
 ParameterPtr
-Slice::Operation::createParameter(const string& name, const TypePtr& type, bool isOutParam, bool isOptional, int tag)
+Slice::Operation::createParameter(const string& name, const TypePtr& type, bool isOptional, int tag)
```

Comment:
Instead of passing `isOutParam` during construction, we now call `setIsOutParam` separately (after construction).

The old grammar rule waited until we saw _everything_ to construct a parameter.
But, the flexibility of the new grammar rules mean that we have less information up-front now.